modules/kuberay-monitoring/main.tf

```diff
@@ -12,14 +12,24 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# google managed prometheus engine
+locals {
+  gmp_namespace = var.autopilot_cluster ? "gke-gmp-system" : "gmp-system"
+}
+
+# Temporary workaround to ensure the GMP webhook is installed before applying PodMonitorings.
+# This dependency can be removed once the GMP webhook is moved from node to KCP.
+resource "null_resource" "wait_for_gmp" {
+  provisioner "local-exec" {
+    command = "while ! kubectl rollout status -n ${local.gmp_namespace} deploy/gmp-operator; do sleep 5; done"
```

Comment:
Does this assume kubectl is available locally? Is that always true?

Reply:
(Closing duplicate comment)